base/inference.jl

```diff
@@ -1439,13 +1489,13 @@ function typeinf(linfo::LambdaStaticData,atypes::Tuple,sparams::Tuple, def, cop)
 
     @label typeinf_top
 
-    old_s1 = nothing
+    typegotoredo = false
```

Comment:
can you check my type_goto algorithm? it's different from yours, but I couldn't entirely follow yours